### PR TITLE
Add connection metrics and track from Libreswan cable driver

### PR DESCRIPTION
**Pull request by maayanf24**
Related to https://github.com/submariner-io/submariner/issues/678

Signed-off-by: Maayan Friedman <maafried@redhat.com>

### CODE REVIEW

File: pkg/cable/metrics.go [hunk 1]
```diff
@@ -55,32 +60,126 @@ var (
...
+		cableDriverLabel:      cableDriverName,
+		localClusterLabel:     localEndpoint.ClusterID,
+		localHostnameLabel:    localEndpoint.Hostname,
+		localEndpointIpLabel:  localEndpoint.PublicIP,
+		remoteClusterLabel:    remoteEndpoint.ClusterID,
+		remoteHostnameLabel:   remoteEndpoint.Hostname,
+		remoteEndpointIpLabel: remoteEndpoint.PublicIP,
+	}
+	txGauge.Delete(labels)
+	rxGauge.Delete(labels)
```
Comment (skitt) [inline pkg/cable/metrics.go:181]: I can’t find the details in the docs right now, but I think that deleting results in the gauge not being exported for those labels, and that might not be what we want for rx/tx. I’ll check tomorrow and update.

File: pkg/cable/metrics.go [hunk 2]
```diff
@@ -55,32 +60,126 @@ var (
...
+			cableDriverLabel,
+			localClusterLabel,
+			localHostnameLabel,
+			localEndpointIpLabel,
 			remoteClusterLabel,
 			remoteHostnameLabel,
+			remoteEndpointIpLabel,
+		},
+	)
+	connectionLatencySecondsGague = prometheus.NewGaugeVec(
```
Comment (skitt) [inline pkg/cable/metrics.go:99]: `Gauge` ;-).

File: pkg/cable/metrics.go [hunk 3]
```diff
@@ -55,32 +60,126 @@ var (
...
+		},
+		[]string{
+			cableDriverLabel,
+			localClusterLabel,
+			localHostnameLabel,
+			localEndpointIpLabel,
+			remoteClusterLabel,
+			remoteHostnameLabel,
+			remoteEndpointIpLabel,
+		},
```
Comment (skitt) [inline pkg/cable/metrics.go:82]: Shouldn’t this have a label for the status too?

**Review by skitt — CHANGES_REQUESTED** [review]

File: pkg/cable/metrics.go (hunk 3 above)
Comment (maayanf24) [reply pkg/cable/metrics.go:82]: then what will be the gauge's value?

File: pkg/cable/metrics.go (hunk 1 above)
Comment (maayanf24) [reply pkg/cable/metrics.go:181]: shouldn't we use same labels for rx/tx gauges as all gauges in this file?

File: pkg/cable/metrics.go (hunk 3 above)
Comment (skitt) [reply pkg/cable/metrics.go:82]: The number of connections per status, same as the operator.

File: pkg/cable/metrics.go [hunk 4]
```diff
@@ -59,28 +63,115 @@ var (
...
+		localEndpointIpLabel:  localEndpoint.PublicIP,
+		remoteClusterLabel:    remoteEndpoint.ClusterID,
+		remoteHostnameLabel:   remoteEndpoint.Hostname,
+		remoteEndpointIpLabel: remoteEndpoint.PublicIP,
+	}
+}
+
+func getTxRxLabels(cableDriverName string, localEndpoint, remoteEndpoint *submv1.EndpointSpec) prometheus.Labels {
+	return prometheus.Labels{
 		cableDriverLabel:    cableDriverName,
```
Comment (skitt) [inline pkg/cable/metrics.go:132]: Regarding your point about whether the tx/rx gauges should have the same labels as the other metrics, you’re right, they should; so we should use `getLabels()` everywhere, and drop `getTxRxLabels()`.

**Review by skitt — CHANGES_REQUESTED** [review]
Looks good, thanks, just a minor tweak (as you suggested) on the tx/rx labels!

**Review by skitt — APPROVED** [review]

**Review by mangelajo — APPROVED** [review]

**Pull request merged**